packages/framework/src/Controller/Admin/ProductController.php

```diff
@@ -372,4 +408,18 @@ public function visibilityAction($productId)
             'domains' => $this->domain->getAll(),
         ]);
     }
+
+    /**
+     * @return bool
+     */
+    protected function productCanBeCreated()
+    {
+        $productCanBeCreated = true;
+
+        if (empty($this->unitFacade->getAll()) || $this->setting->get(Setting::DEFAULT_UNIT) === 0 || empty($this->availabilityFacade->getAll())) {
+            $productCanBeCreated = false;
```

Comment:
unnecessary variable, please change to `return false` and after `if` statement `return true`

Reply:
updated